File: solutions/cpp/Problem268.cpp
#pragma once
/*
268. Missing Number

Given an array containing n distinct numbers taken from 0, 1, 2, ..., n, find the one that is missing from the array.
*/
#include <vector>


class Solution {
public:
	int missingNumber(std::vector<int>& nums) {
		int res = (nums.size() + 1) * nums.size() / 2;
		for (auto &i : nums) {
			res -= i;
		}
		return res;
	}
};

File: solutions/python3/problem268.py
# -*- coding: utf-8 -*-
"""
268. Missing Number

Given an array nums containing n distinct numbers in the range [0, n],
return the only number in the range that is missing from the array.

Follow up: Could you implement a solution using only O(1) extra space complexity and O(n) runtime complexity?

Constraints:

n == nums.length
1 <= n <= 104
0 <= nums[i] <= n
All the numbers of nums are unique.
"""


class Solution:
    def missingNumber(self, nums):
        res = (len(nums) * (len(nums) + 1)) // 2
        for num in nums:
            res -= num

        return res
